File: examples/FGM/ip_design/src/foo_user.m
%% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% icl::protoip
% Author: asuardi <https://github.com/asuardi>
% Date: November - 2014
%% %%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%


function [laff_out_out_int] = foo_user(project_name,laff_in_in_int)


	% load project configuration parameters: input and output vectors (name, size, type, NUM_TEST, TYPE_TEST)
	load_configuration_parameters(project_name);

	%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
	% compute with Matlab and save in a file simulation results laff_out_out_int
	for i=1:LAFF_OUT_OUT_LENGTH
		laff_out_out_int(i)=0;
		for i_laff_in = 1:LAFF_IN_IN_LENGTH
			laff_out_out_int(i)=laff_out_out_int(i) + laff_in_in_int(i_laff_in);
		end
	end

end

File: examples/FGM/ip_design/src/foo_user.cpp
/* 
* icl::protoip
* Author: asuardi <https://github.com/asuardi>
* Date: November - 2014
*/


#include "foo_data.h"


void foo_user(  data_t_laff_in_in laff_in_in_int[LAFF_IN_IN_LENGTH],
				data_t_laff_out_out laff_out_out_int[LAFF_OUT_OUT_LENGTH])
{

	///////////////////////////////////////
	//ADD USER algorithm here below:
	//(this is an example)
	alg_0 : for(int i = 0; i <LAFF_OUT_OUT_LENGTH; i++)
	{
		laff_out_out_int[i]=0;
		loop_0_laff_in : for(int i_laff_in = 0; i_laff_in <LAFF_IN_IN_LENGTH; i_laff_in++)
		{
			laff_out_out_int[i]=laff_out_out_int[i] + (data_t_laff_out_out)laff_in_in_int[i_laff_in];
		}
	}

}
